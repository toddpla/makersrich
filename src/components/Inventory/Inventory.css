#inventory-box {
  width: 800px;
  height: 640px;
  background-image: url('/assets/quizzical.png');
  border: 7px solid #e1c699;
}

#inventory-list-box {
  text-align: center;
  width: 400px;
  height: 640px;
  line-height: 1.5em;
  position: absolute;
  border-right: 7px solid #e1c699
}

.inventory-list-item {
  display: inline-block;
  position: relative;
  padding-top: 10px;
  width: 50px;
  height: 50px;
  float: left;
}

.inventory-item-message {
  text-align: center;
  left: 427px;
  height: 640px;
  width: 300px;
  position: absolute;
  padding-left: 50px;
  padding-right: 50px;
  line-height: 1.5em;
}

.ruby {
  background-image: url('/assets/ruby.png');
  width: 40px;
  height: 40px;
  background-repeat: no-repeat;
  background-size: 40px;
  background-position: center;
}

.bean {
  background-image: url('/assets/bean.png');
  width: 40px;
  height: 40px;
  background-repeat: no-repeat;
  background-size: 40px;
  background-position: center;
}

.key {
  background-image: url('/assets/key.png');
  width: 40px;
  height: 40px;
  background-repeat: no-repeat;
  background-position: center;
  background-size: 40px;
}

.inventory-selected-item {
  position: absolute;
  top:230px;
  right: 162px;
  width: 70px;
  height: 70px;
  background-repeat: no-repeat;
  background-size: 70px;
  background-position: center;
}
