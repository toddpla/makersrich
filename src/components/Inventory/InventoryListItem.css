.inventory-list-item {
  border: 5px solid black
}
