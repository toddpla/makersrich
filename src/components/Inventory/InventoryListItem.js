import React, { Component } from 'react';

class InventoryListItem extends Component {

  changeMessage = () => {
    if (this.props.item.properties && this.props.item.properties[0].value) {
      this.props.handleMessage(this.props.item.properties[0].value)
    } else {
      this.props.handleMessage(`This is a ${this.props.item.type}`)
    }
  }

  render() {
    console.log(this.props.item);
    return (
      <div className="inventory-list-item" onClick={this.changeMessage}>
        <div className={this.props.item.type}></div>
      </div>
    );
  }

}

export default InventoryListItem;
