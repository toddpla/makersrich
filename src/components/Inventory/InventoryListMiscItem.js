import React, { Component } from 'react';

class InventoryListMiscItem extends Component {

  changeMessage = () => {
      this.props.handleMessage(this.props.item.properties.message)
  }

  render() {
    return (
      <div className="inventory-list-item" onClick={this.changeMessage}>
        <div className={`miscellaneous-item ${this.props.item.properties.type}`}></div>
      </div>
    );
  }

}

export default InventoryListMiscItem;
