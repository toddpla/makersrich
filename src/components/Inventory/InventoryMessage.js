import React, { Component } from 'react';

class InventoryMessage extends Component {

  handleMessage = () => {
    if(this.props.selected === undefined){
      return
    }
    if(this.props.selected.type === 'miscellaneous'){
      return console.log(this.props.selected);
      // return this.props.selected.properties[0].value
    }
    if (this.props.selected.properties && this.props.selected.properties[0].value) {
       return this.props.selected.properties[0].value
    } else {
      return `This is a ${this.props.selected.type}`
    }
  }

  render() {
    return (
      <div id="inventory-item-message">
        <h1>Information</h1>
        <p>{this.handleMessage()}</p>
      </div>
    );
  }

}

export default InventoryMessage;
