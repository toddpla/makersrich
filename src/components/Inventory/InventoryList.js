import React, { Component } from 'react';
import InventoryListItem from './InventoryListItem'

class InventoryList extends Component {

  render() {

    return (
      <div>
        {this.props.inventory.ruby.map((item, i) => <InventoryListItem key={i} item={item} />)}
        {this.props.inventory.bean.map((item, i) => <InventoryListItem key={i} item={item} />)}
        {this.props.inventory.key.map((item, i) => <InventoryListItem key={i} item={item} />)}
      </div>
    );
  }

}

export default InventoryList;

// let newArr = [];
//
// while(allItems.length) newArr.push(allItems.splice(0,2));

// <table>
//   <tbody>
//     {newArr.map((row, i) => (
//       <tr>
//         {row.map((cell, k) => (
//           <td>
//             {cell}
//           </td>
//         ))}
//       </tr>
//     ))}
//   </tbody>
// </table>
