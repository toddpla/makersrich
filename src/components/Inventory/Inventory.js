import React from 'react'
import { connect } from 'react-redux'

import InventoryList from './InventoryList'
import InventoryMessage from './InventoryMessage'

export class Inventory extends React.Component {
  constructor(props) {
    super(props)
    this.state = {
      message: undefined
    }
  }

  handleMessage = (message) => {
    this.setState({message: message})
  }

  render() {
    return (
      <div id="inventory-box">
        <InventoryList inventory={this.props.player.inventory} handleMessage={this.handleMessage}/>
        <InventoryMessage message={this.state.message} />
      </div>
    )
  }
}

const mapStateToProps = (state) => ({
  player: state.auth
})

export default connect(mapStateToProps)(Inventory)
