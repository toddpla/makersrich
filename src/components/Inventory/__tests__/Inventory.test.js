import React from 'react';
import { shallow } from 'enzyme'
import { Inventory } from '../Inventory'
import player from '../../../test/fixtures/playerWithInventory'

let wrapper

beforeEach(function() {
  wrapper = shallow(
    <Inventory
      player={player}
    />
  )
});

test('it renders Inventory', () => {
  expect(wrapper).toMatchSnapshot()
})

test('#handleMessage changes message state', () => {
  const instance = wrapper.instance()
  expect(wrapper.state.message).toEqual(undefined)
  instance.handleMessage("message")
  expect(instance.state.message).toEqual('message')
})
