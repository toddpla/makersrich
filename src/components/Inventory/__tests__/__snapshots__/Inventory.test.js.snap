// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`it renders Inventory 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Inventory
    player={
      Object {
        "cash": 0,
        "displayName": "Frankie",
        "inventory": Object {
          "bean": Array [
            Object {
              "properties": Array [
                Object {
                  "value": "This is a bean",
                },
              ],
              "type": "bean",
            },
          ],
          "key": Array [
            Object {
              "properties": Array [
                Object {
                  "value": "This is a key",
                },
              ],
              "type": "key",
            },
          ],
          "ruby": Array [
            Object {
              "properties": Array [
                Object {
                  "value": "This is a ruby",
                },
              ],
              "type": "ruby",
            },
          ],
        },
        "left": 0,
        "level": 1,
        "state": "online",
        "top": 0,
        "uid": "1",
      }
    }
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <InventoryList
          handleMessage={[Function]}
          inventory={
            Object {
              "bean": Array [
                Object {
                  "properties": Array [
                    Object {
                      "value": "This is a bean",
                    },
                  ],
                  "type": "bean",
                },
              ],
              "key": Array [
                Object {
                  "properties": Array [
                    Object {
                      "value": "This is a key",
                    },
                  ],
                  "type": "key",
                },
              ],
              "ruby": Array [
                Object {
                  "properties": Array [
                    Object {
                      "value": "This is a ruby",
                    },
                  ],
                  "type": "ruby",
                },
              ],
            }
          }
        />,
        <InventoryMessage />,
      ],
      "id": "inventory-box",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "handleMessage": [Function],
          "inventory": Object {
            "bean": Array [
              Object {
                "properties": Array [
                  Object {
                    "value": "This is a bean",
                  },
                ],
                "type": "bean",
              },
            ],
            "key": Array [
              Object {
                "properties": Array [
                  Object {
                    "value": "This is a key",
                  },
                ],
                "type": "key",
              },
            ],
            "ruby": Array [
              Object {
                "properties": Array [
                  Object {
                    "value": "This is a ruby",
                  },
                ],
                "type": "ruby",
              },
            ],
          },
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "message": undefined,
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <InventoryList
            handleMessage={[Function]}
            inventory={
              Object {
                "bean": Array [
                  Object {
                    "properties": Array [
                      Object {
                        "value": "This is a bean",
                      },
                    ],
                    "type": "bean",
                  },
                ],
                "key": Array [
                  Object {
                    "properties": Array [
                      Object {
                        "value": "This is a key",
                      },
                    ],
                    "type": "key",
                  },
                ],
                "ruby": Array [
                  Object {
                    "properties": Array [
                      Object {
                        "value": "This is a ruby",
                      },
                    ],
                    "type": "ruby",
                  },
                ],
              }
            }
          />,
          <InventoryMessage />,
        ],
        "id": "inventory-box",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "handleMessage": [Function],
            "inventory": Object {
              "bean": Array [
                Object {
                  "properties": Array [
                    Object {
                      "value": "This is a bean",
                    },
                  ],
                  "type": "bean",
                },
              ],
              "key": Array [
                Object {
                  "properties": Array [
                    Object {
                      "value": "This is a key",
                    },
                  ],
                  "type": "key",
                },
              ],
              "ruby": Array [
                Object {
                  "properties": Array [
                    Object {
                      "value": "This is a ruby",
                    },
                  ],
                  "type": "ruby",
                },
              ],
            },
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "message": undefined,
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
